src/status_im/subs/keycard.cljs

```diff
@@ -6,6 +6,40 @@
  (fn [db]
    (not (nil? (get-in db [:profile/profile :keycard-pairing])))))
 
+(rf/reg-sub
+ :keycard/keycard-profile
+ :<- [:profile/name]
+ :<- [:profile/image]
+ :<- [:profile/customization-color]
+ (fn [[profile-name profile-image customization-color]]
+   {:profile-name        profile-name
+    :profile-image       profile-image
+    :customization-color customization-color}))
+
+(rf/reg-sub
+ :keycard/keypairs-keycards
+ :<- [:wallet/keypairs-list]
+ (fn [keypairs]
+   (reduce (fn [keycards keypair]
+             (if (and (= :profile (:type keypair))
+                      (:keycards keypair))
+               (concat keycards (:keycards keypair))
```

Comment:
Untested code: I think you can split the filtering from the reduction to improve readability. I'm not using the transducer style, but `mapcat` can return a transducer if you prefer that.

```clojure
(->> keypairs
     (filter (fn [{:keys [type keycards]}]
               (and (= :profile type) keycards)))
     (mapcat :keycards))
```

Once the filtering is separate we can even see more clearly we can have a separate predicate, like `(filter profile-keypair?)`.